apps/studio/src/pages/sites/[siteId]/index.tsx

```diff
@@ -22,12 +26,22 @@ import { CreatePageModal } from "~/features/editing-experience/components/Create
 import { MoveResourceModal } from "~/features/editing-experience/components/MoveResourceModal"
 import { useQueryParse } from "~/hooks/useQueryParse"
 import { type NextPageWithLayout } from "~/lib/types"
+import { buildPermissionsFor } from "~/server/modules/permissions/permissions.util"
```

Comment:
feels weird to import a server module on FE

rather BE should be returning a fully-formed response than FE doing this

Reply:
this is technically a util method - i agree iwth you and i tried sending a json of the `builder.rules` but i ran into type issues generating the ability and went down this route.

if it's the import, i can shift it into a top-level `/utils/permissions` file? i think the important thing for me is using a shared method to generate the ability.